library/network/src/modules/SuSEFirewall.rb

```diff
@@ -648,6 +648,13 @@ def SaveAndRestartService
       true
     end
 
+    def morph_to(backend_sym)
+      # Sometimes it's more convenient to send the backend string name.
+      backend_sym = @@firewall_backends.select { |k, v| v == backend_sym }.keys.first unless backend_sym.is_a?(Symbol)
+      Yast.send(:remove_const, 'SuSEFirewall')
+      Yast.const_set('SuSEFirewall', self.class.superclass.create(backend_sym))
```

Comment:
Resetting constants is pretty tough stuff. I'll see whether we can use a more conventional approach.